src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api.UnitTests/Services/QueryStringParameterBuilderTests.cs

```diff
@@ -0,0 +1,227 @@
+        }
+        else
+        {
+            fromDateResult.Should().BeNull();
+        }
+    }
+
+    [TestCase(null)]
+    [TestCase("2030-06-01")]
+    public void Builder_ConstructParameters_ToDate(DateTime? toDate)
+    {
+        var model = new GetCalendarEventsRequestModel
+        {
+            RequestedByMemberId = Guid.NewGuid(),
+            ToDate = toDate
+        };
+        var parameters = QueryStringParameterBuilder.BuildQueryStringParameters(model);
+
+
+        parameters.TryGetValue("toDate", out var toDateResult);
+        if (toDate != null)
+        {
+            toDateResult![0].Should().Be(toDate?.ToString("yyyy-MM-dd"));
+        }
+        else
+        {
+            toDateResult.Should().BeNull();
+        }
+    }
+
+    [TestCase(null)]
+    [TestCase(EventFormat.Hybrid)]
+    public void Builder_ConstructParameters_EventFormats(EventFormat? eventFormat)
+    {
+        var model = new GetCalendarEventsRequestModel
+        {
+            RequestedByMemberId = Guid.NewGuid(),
+        };
+
+        if (eventFormat != null)
+        {
+            model.EventFormat = new List<EventFormat> { eventFormat.Value };
+        }
+
+        var parameters = QueryStringParameterBuilder.BuildQueryStringParameters(model);
+
+        parameters.TryGetValue("eventFormat", out var eventFormatResult);
+        if (eventFormatResult != null)
+        {
+            eventFormatResult![0].Should().BeEquivalentTo(model.EventFormat[0].ToString());
+        }
+        else
+        {
+            eventFormatResult.Should().BeNull();
+        }
+    }
+
+    [TestCase(null)]
+    [TestCase(1)]
+    public void Builder_ConstructParameters_CalendarIds(int? calendarId)
+    {
+        var model = new GetCalendarEventsRequestModel
+        {
+            RequestedByMemberId = Guid.NewGuid(),
+        };
+
+        if (calendarId != null)
+        {
+            model.CalendarId = new List<int> { calendarId.Value };
+        }
+
+        var parameters = QueryStringParameterBuilder.BuildQueryStringParameters(model);
+
+        parameters.TryGetValue("calendarId", out var calendarIdResult);
+        if (calendarIdResult != null)
+        {
+            calendarIdResult![0].Should().BeEquivalentTo(model.CalendarId[0].ToString());
+        }
+        else
+        {
+            calendarIdResult.Should().BeNull();
+        }
+    }
+
+    [TestCase(null)]
+    [TestCase(1)]
+    public void Builder_ConstructParameters_RegionIds(int? regionId)
+    {
+        var model = new GetCalendarEventsRequestModel
+        {
+            RequestedByMemberId = Guid.NewGuid(),
+        };
+
+        if (regionId != null)
+        {
+            model.RegionId = new List<int> { regionId.Value };
+        }
+
+        var parameters = QueryStringParameterBuilder.BuildQueryStringParameters(model);
+
+        parameters.TryGetValue("regionId", out var regionIdResult);
+        if (regionIdResult != null)
```

Comment:
Same issue here as calendarId

Reply:
But the parameter list won't have a match if the regionId is a list of 0, so the regionIdResult is null in that case.